text/0000-drop-types-in-const.md

```diff
@@ -0,0 +1,68 @@
+- Feature Name: `drop_types_in_const`
+- Start Date: 2016-01-01
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow types with destructors to be used in `static` items and in `const` functions, as long as the destructor never needs to run in const context.
+
+# Motivation
+[motivation]: #motivation
+
+Most collection types do not allocate any memory when constructed empty. With the change to make leaking safe, the restriction on `static` items with destructors
```

Comment:
You mean *some*. @Gankro, I believe, explicitly expressed the intent to specify that we only will have allocation less `new` when it doesn’t hurt performance/implementation complexity or something like that?